cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/program/MessagingProgramStateWriter.java

```diff
@@ -68,12 +68,12 @@ public void start(ProgramRunId programRunId, ProgramOptions programOptions, @Nul
                     ArtifactId artifactId) {
     long startTime = RunIds.getTime(RunIds.fromString(programRunId.getRun()), TimeUnit.MILLISECONDS);
```

Comment:
Seems like this is not needed as well? If there is need for validation, should validate at the consumer end in order to skip invalid messages. The writer shouldn't need to validate.